schedserver/task.go

```diff
@@ -2,39 +2,123 @@ package schedserver
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	api "github.com/arthurfabre/scheduler/schedapi"
 	"github.com/arthurfabre/scheduler/schedserver/pb"
 	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/clientv3util"
 	"github.com/golang/protobuf/proto"
+	"github.com/satori/go.uuid"
 	"log"
-	"strconv"
+	"strings"
 	"time"
 )
 
+// Format strings for keys. Don't include the last %s for the Task UUID
+// See README/#ETCD Key Schema
 const (
-	keySep       = "/"
-	taskPrefix   = "task" + keySep
-	statusPrefix = taskPrefix + "status" + keySep
+	taskFmt     = "task/"
+	statusFmt   = taskFmt + "status/"
+	queuedFmt   = statusFmt + "queued/"
+	runningFmt  = statusFmt + "running/%s/"
+	completeFmt = statusFmt + "complete/%d/"
+	canceledFmt = statusFmt + "canceled/%d/"
```

Comment:
I think some of this code would be much easier to understand, if these were considered prefixes, and matched with helper functions. So something more along the lines of:

```
const ( 
  completePrefix = "/task/status/running"
)

func getCompletedKey(uuid string) string {
  return fmt.Sprint(completePrefix, uuid)
}
```

So then instead of having something like
```
         case *api.TaskStatus_Running_:
		return key(runningFmt, status.GetRunning().NodeId.Uuid)
 	case *api.TaskStatus_Complete_:
		return key(completeFmt, status.GetComplete().Epoch)
```

It becomes
```
         case *api.TaskStatus_Running_:
		return getRunningKey(status.GetRunning().NodeId.Uuid)
 	case *api.TaskStatus_Complete_:
		return getCompleteKey(status.GetComplete().Epoch)
```